### PR TITLE
App menu

### PR DESCRIPTION
**Pull request by AdamDotCom**
Related PR: https://github.com/zer0-os/zOS-feed/pull/2

## Screen Capture

zOS-feed

### CODE REVIEW

File: src/store/apps/index.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const initialState: AppsState = {
+  selectedApp: config.defaultApp,
+};
+
+const slice = createSlice({
+  name: 'apps',
+  initialState,
+  reducers: {
+    receive: (state, action: PayloadAction<string>) => {
+      state.selectedApp = action.payload;
```
Comment (AdamDotCom) [inline src/store/apps/index.ts:28]: I'm conflicted. The zns state containing the route also seems like an appropriate place to store the selected app. It seems unlikely that we'll access the ZNS route without the app. What are your thoughts?

File: src/zns-route-connect.tsx [hunk 2]
```diff
@@ -19,24 +21,34 @@ export class Container extends React.Component<Properties> {
   }
 
   static mapActions(_props: Properties): Partial<Properties> {
-    return { setRoute };
+    return { setRoute, setSelectedApp };
   }
 
   componentDidMount() {
     this.props.setRoute(this.extractRouteFromProps());
+    this.props.setSelectedApp(this.extractAppFromProps());
```
Comment (AdamDotCom) [inline src/zns-route-connect.tsx:29]: Is this double saga call a performance issue?

File: src/components/address-bar/container.tsx [hunk 3]
```diff
@@ -14,13 +14,14 @@ export interface Properties extends PublicProperties {
   history: History;
   route: string;
   deepestVisitedRoute: string;
+  app: string;
```
Comment (mrval5) [inline src/components/address-bar/container.tsx:17]: If I'm not mistaken the type should be `Apps`

Comment (AdamDotCom) [reply src/components/address-bar/container.tsx:17]: Thanks!

**Review by mrval5 — DISMISSED** [review]

**Pull request merged**